agent.go

```diff
@@ -331,6 +331,13 @@ func NewAgent(config *AgentConfig) (*Agent, error) {
 	if err != nil {
 		return nil, err
 	}
+	closeMDNSConn := func() {
```

Comment:
Before we leaked mDNS connections when errors happened, nice little find from improved tooling :)